umm-spec-lib/src/cmr/umm_spec/test/iso19115_expected_conversion.clj

```diff
@@ -321,6 +322,58 @@
         role (:Roles data-center)]
    (assoc data-center :Roles [role]))))
 
+(defn- update-contact-persons-from-collection-citation
+  "CollectionCitation introduces additional contact persons from creator, editor and publisher.
+   They need to be added to the ContactPersons in the expected umm after converting the umm
+   to the xml and back to umm.  Returns the updated ContactPersons"
+  [contact-persons collection-citation]
+  (let [creator (:Creator collection-citation)
+        editor (:Editor collection-citation)
+        publisher (:Publisher collection-citation)
+        creator-contact-person (when creator
```

Comment:
I think this `when` and `let` logic could be pretty nicely pulled out into a helper function above. I think generally we want to avoid having any kind of conditionals in `let` bindings